### ISSUE TITLE
[BAM: Bottleneck Attention Module] Implementation and paper is dissimilar

### ISSUE THREAD

**Issue by vietnamican** [osmr/imgclsmob#78]
Accoding to the paper of [BAM](https://arxiv.org/pdf/1807.06514v2.pdf), feature maps of ChannelGate and SpatialGate is summation before sigmoid. But in this code: https://github.com/osmr/imgclsmob/blob/30d63bae444a71c07f61f87ce66c8dad90f214e7/pytorch/pytorchcv/models/bamresnet.py#L156
An element-wise multiplication is applied which is perform not as good as summation in Table 2(b) due to "efficient gradient flow"
Can you explain me about this change?

**Comment by osmr** [comment]
Interesting. I took already trained weights from the official repository (https://github.com/Jongchan/attention-module). They are quite consistent with this code (for comparison: https://github.com/Jongchan/attention-module/blob/master/MODELS/bam.py) in terms of quality. It may be worth re-train model with the sum and compare the results.

**Comment by vietnamican** [comment]
Oh, I saw it! But I just use BamBlock into my work. I'll give it a try and compare the results if I have time. Thank you for your help!
